log/example_test.go

```diff
@@ -96,7 +98,40 @@ func Example_debugInfo() {
 	logger.Log("call", "third")
 
 	// Output:
-	// time=2015-02-03T10:00:01Z caller=example_test.go:91 call=first
-	// time=2015-02-03T10:00:02Z caller=example_test.go:92 call=second
-	// time=2015-02-03T10:00:03Z caller=example_test.go:96 call=third
+	// time=2015-02-03T10:00:01Z caller=example_test.go:93 call=first
+	// time=2015-02-03T10:00:02Z caller=example_test.go:94 call=second
+	// time=2015-02-03T10:00:03Z caller=example_test.go:98 call=third
+}
+
+func Example_syncWriter() {
+	w := log.NewSyncWriter(os.Stdout)
+	logger := log.NewLogfmtLogger(w)
+
+	type Task struct {
+		ID int
+	}
+
+	var wg sync.WaitGroup
+
+	RunTask := func(task Task, logger log.Logger) {
+		logger.Log("taskID", task.ID, "event", "starting task")
+
+		time.Sleep(time.Duration(rand.Intn(200)) * time.Millisecond)
+
+		logger.Log("taskID", task.ID, "event", "task complete")
+		wg.Done()
+	}
+
+	wg.Add(2)
+
+	go RunTask(Task{ID: 1}, logger)
+	go RunTask(Task{ID: 2}, logger)
+
+	wg.Wait()
+
+	// Unordered output:
```

Comment:
Glad I could help.. I had basically the exact same issue with a library I wrote a while back so I figured it would be useful.. the deploy became complicated because godoc has to be backwards compatible so that kind of became a crap show for deployments. =(